frontend/client/src/screens/CodeValidations.js

```diff
@@ -26,6 +28,19 @@ const CodeValidationsBase = observer((props) => {
 
   let confirmedToast = createRef()
 
+  useEffect(() => {
+    updateButtonDisabled()
+  })
+
+  // Updates the buttonDisabled variable state based on other state variables
+  const updateButtonDisabled = () => {
+    if (needsReset || testType === '' || dateInvalid) {
```

Comment:
I believe one final option is needed in this conditional in order to prevent a situation where the generate code button is enabled by clicking a diagnosis radio but NOT filling out a date:

`|| (testType !== '' && testDate === '')`

So, the if statement on line 37 would read:

`if (needsReset || testType === '' || dateInvalid || (testType !== '' && testDate === ''))`

I tested this locally and I believe it protects our state the way we intend.